src/test/kotlin/cacophony/semantic/analysis/StaticFunctionRelationsTest.kt

```diff
@@ -152,13 +185,130 @@ class StaticFunctionRelationsTest {
                         StaticFunctionRelations(
                             program,
                             1,
-                            setOf(varA, varB, varC),
+                            setOf(aVariable, bVariable, cVariable),
                             emptySet(),
                         ),
                 ),
             )
     }
 
+    @Test
+    fun `should find struct variables`() {
+        // given
+        //       s
+        //      / \
+        //     /   \
+        //    a     b
+        //    |    / \
+        //    c   d   e
+        //        |
+        //        f
+        //
+        //
+        // f => ( let s, s.a = s.b.d)
+        val sDeclaration = variableDeclaration("s", Empty(mockRange()))
+        val aDeclaration = variableDeclaration("a", Empty(mockRange()))
```

Comment:
our compiler doesn't create any of these declarations besides sDeclaration and we don't need to create them as well, removing them would make code clearer (similarly in VarUseTypeAnalysisTest)